app/src/main/java/cc/pogoda/mobile/meteosystem/adapter/WeatherStationRecyclerViewAdapter.java

```diff
@@ -62,77 +68,95 @@ public WeatherStationRecyclerViewAdapter(List<WeatherStation> stations, AppCompa
 
     @NonNull
     @Override
-    public AllStationsActRecyclerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
         Context context = parent.getContext();
         LayoutInflater inflater = LayoutInflater.from(context);
-
         View view;
 
-        // check the call reason
-        if (reason.equals(ParceableFavsCallReason.Reason.FAVOURITES)) {
-            // inflate custom layout
-            view = inflater.inflate(R.layout.activity_favourites_linear_layout_data, parent, false);
-        }
-        else {
-            // Inflate the custom layout without current data
-            view = inflater.inflate(R.layout.activity_all_stations_linear_layout, parent, false);
+        switch (viewType){
+            case VIEW_TYPE_OBJECT:
+                // check the call reason
+                if (reason.equals(ParceableFavsCallReason.Reason.FAVOURITES)) {
+                    // inflate custom layout
+                    view = inflater.inflate(R.layout.activity_favourites_linear_layout_data,
+                            parent, false);
+                }
+                else {
+                    // Inflate the custom layout without current data
+                    view = inflater.inflate(R.layout.activity_all_stations_linear_layout,
+                            parent, false);
+                }
+                return new AllStationsActRecyclerViewHolder(view, reason);
+
+            case VIEW_TYPE_EMPTY_LIST:
+            default:
+                if (reason.equals(ParceableFavsCallReason.Reason.FAVOURITES)) {
```

Comment:
This could be imporved. THis View adapter already have 2 functions which do not follow SOLID principles. We can split it: one for AllStationsList and one for Favourites list.

Reply:
same for ViewHolders